src/app-layer-enip-common.c

```diff
@@ -91,10 +91,12 @@ static int ENIPExtractUint32(uint32_t *res, const uint8_t *input, uint16_t *offs
         return 0;
     }
 
-    ByteExtractUint32(res, BYTE_LITTLE_ENDIAN, sizeof(uint32_t),
-            (const uint8_t *) (input + *offset));
-    *offset += sizeof(uint32_t);
-    return 1;
+    if(ByteExtractUint32(res, BYTE_LITTLE_ENDIAN, sizeof(uint32_t),
```

Comment:
The warnings are generated due to the output of this function being never actually used in the call.

The purpose of the caller function `ENIPExtractUint32` seems to get satisfied even without a return value being called and used from the `ByteExtractUint32` function.

The only thing that is making sense to me is adding an `if{}` like above and function returning success if bytes get extracted i.e `-1` is not returned.

Is this the right approach?

Reply:
Doesn't look like it is. In other anomaly conditions, we return `0` (e.g. not enough input bytes), so if `ByteExtractUint32` fails we should probably do the same.

Reply:
I did not quite understand I think. You are suggesting that here we should return 0 if `ByteExtractUint32 ` fails?

Reply:
Yes